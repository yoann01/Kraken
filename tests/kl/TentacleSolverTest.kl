
require Kraken;

operator entry(){
  TentacleSolver solver();

  Boolean drawDebug = true;
  Scalar rigScale = 1.0;

  Mat44 fkcontrols[];
  fkcontrols.resize(6);

  // Outputs
  Mat44 pose[];
  pose.resize(6);

  Mat44 tentacleEnd;

  // TODO: Finish setting up test values and test with simulation.

  // solver.solve(
  //   drawDebug,
  //   rigScale,
  //   time,
  //   ikblend,
  //   waveLength_Y,
  //   waveAmplitude_Y,
  //   waveFrequency_Y,
  //   waveLength_Z,
  //   waveAmplitude_Z,
  //   waveFrequency_Z,
  //   tipBias,

  //   springStrength,
  //   dampening,
  //   simulationWeight,
  //   softLimitBounds,

  //   chainBase,
  //   ikgoal,
  //   fkcontrols,
  //   tipBoneLen,
  //   pose,
  //   tentacleEnd
  // )

  report(pose);
}
