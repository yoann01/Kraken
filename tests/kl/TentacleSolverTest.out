[{row0:{x:+1.0,y:+0.0,z:+0.0,t:+0.0},row1:{x:+0.0,y:+1.0,z:+0.0,t:+0.0},row2:{x:+0.0,y:+0.0,z:+1.0,t:+0.0},row3:{x:+0.0,y:+0.0,z:+0.0,t:+1.0}},{row0:{x:+1.0,y:+0.0,z:+0.0,t:+0.0},row1:{x:+0.0,y:+1.0,z:+0.0,t:+0.0},row2:{x:+0.0,y:+0.0,z:+1.0,t:+0.0},row3:{x:+0.0,y:+0.0,z:+0.0,t:+1.0}},{row0:{x:+1.0,y:+0.0,z:+0.0,t:+0.0},row1:{x:+0.0,y:+1.0,z:+0.0,t:+0.0},row2:{x:+0.0,y:+0.0,z:+1.0,t:+0.0},row3:{x:+0.0,y:+0.0,z:+0.0,t:+1.0}},{row0:{x:+1.0,y:+0.0,z:+0.0,t:+0.0},row1:{x:+0.0,y:+1.0,z:+0.0,t:+0.0},row2:{x:+0.0,y:+0.0,z:+1.0,t:+0.0},row3:{x:+0.0,y:+0.0,z:+0.0,t:+1.0}},{row0:{x:+1.0,y:+0.0,z:+0.0,t:+0.0},row1:{x:+0.0,y:+1.0,z:+0.0,t:+0.0},row2:{x:+0.0,y:+0.0,z:+1.0,t:+0.0},row3:{x:+0.0,y:+0.0,z:+0.0,t:+1.0}},{row0:{x:+1.0,y:+0.0,z:+0.0,t:+0.0},row1:{x:+0.0,y:+1.0,z:+0.0,t:+0.0},row2:{x:+0.0,y:+0.0,z:+1.0,t:+0.0},row3:{x:+0.0,y:+0.0,z:+0.0,t:+1.0}}]